### PR TITLE
App issues and Vultisig update

### PR DESCRIPTION
**Pull request by proofmancer**
Fixes:
- [Update vultisig integration to support thorchain](https://linear.app/project-chaos/issue/YLD-103/update-vultisig-integration-to-support-thorchain) (Left for after release)
- [Unable to select WalletConnect](https://linear.app/project-chaos/issue/YLD-98/unable-to-select-walletconnect-option)
- [Phantom appears as detected](https://linear.app/project-chaos/issue/YLD-93/phantom-appears-as-detected-wallet-when-not-installed)
- [Solana Icon Dissapears](https://linear.app/project-chaos/issue/YLD-91/solana-icon-disappears-when-clicking-on-hardware-wallets)
- [Close modal when HW wallet connected](https://linear.app/project-chaos/issue/YLD-94/close-wallet-modal-after-successfully-connecting-ledger)
- Other UX/UI updates

### CODE REVIEW

File: app/app/components/modals/Wallet/ChainSelector.tsx [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
     <div className="flex flex-col gap-4">
-      <h3 className="text-base text-neutral-900 font-medium font-gt-america">
-        Select Chain
-      </h3>
+      <div className="flex items-center justify-between">
+        <h3 className="text-base text-neutral-900 font-medium font-gt-america">
+          Select Chain
+        </h3>
+        <Button
+          label={selectedChain.length !== 9 ? "select" : "deselect all"}
```
Comment (MattNguyen) [inline app/app/components/modals/Wallet/ChainSelector.tsx]: titlecase please

File: app/app/modal.tsx [hunk 2]
```diff
@@ -18,11 +18,11 @@ export default function Modal({
 
   return (
     <div
-      className="fixed z-[1000] inset-0 w-full h-full overflow-y-auto bg-[rgb(0,0,0,0.5)] flex items-center justify-center"
+      className="fixed z-[1000] inset-0 w-full h-full bg-[rgb(0,0,0,0.5)] flex items-center justify-center"
       onClick={onClose}
     >
       <div
-        className="bg-white-radial backdrop-blur-md border-4 border-white relative z-20 max-w-lg w-[95%] mx-auto rounded-xl p-7"
+        className="bg-neutral-100 backdrop-blur-md border-4 border-white h-[95vh] relative z-20 max-w-lg w-[95%] mx-auto rounded-xl p-7 overflow-y-auto"
```
Comment (MattNguyen) [inline app/app/modal.tsx]: why h-95vh? makes the modal look pretty tall on large displays. prefer if this was max height but maybe we just take it out

File: app/hooks/useWalletConnection.ts [hunk 3]
```diff
@@ -54,7 +54,7 @@ export function useWalletConnection(
       };
 
       const detectedWalletForChain = detectedWallets.find((w) => {
-        const identifier = chainIdentifiers[selectedChain];
+        const identifier = chainIdentifiers["solana"];
```
Comment (MattNguyen) [inline app/hooks/useWalletConnection.ts]: seems like a mistake hardcoding `solana` here?

**Review by MattNguyen — CHANGES_REQUESTED** [review]
can you also run `pnpm lint:fix` please

**Comment by MattNguyen** [comment]
@sstefdev 

phantom still appears as detected even though i have it disabled

image

**Comment by MattNguyen** [comment]
After I connect with WalletConnect, the WalletConnect qr code modal closes but nothing happens.

Get this in the dev console:

image

**Comment by MattNguyen** [comment]
Have you verified vulticonnect works with thorchain on a local build of vulticonnect? https://github.com/vultisig/vulticonnect/pull/71 hasn't been merged so we should disable vultisig<>thorchain for now until it's released to public.

**Comment by proofmancer** [comment]
> Have you verified vulticonnect works with thorchain on a local build of vulticonnect? [vultisig/vulticonnect#71](https://github.com/vultisig/vulticonnect/pull/71) hasn't been merged so we should disable vultisig<>thorchain for now until it's released to public.

Yes I have tested it with the PR version that it was shared with us, but I will remove it till it is stable version

**Comment by proofmancer** [comment]
> @sstefdev
> 
> phantom still appears as detected even though i have it disabled
> 
> image

@MattNguyen please pull the latest changes and try again

**Comment by proofmancer** [comment]
> After I connect with WalletConnect, the WalletConnect qr code modal closes but nothing happens.
> 
> Get this in the dev console:
> 
> image

This is now fixed again with the latest changes

**Review by MattNguyen — APPROVED** [review]

**Pull request merged**